torch/distributed/pipelining/schedules.py

```diff
@@ -2185,6 +2238,9 @@ def _perform_action(action: _Action) -> None:
                     mb_index,
                     last_backward=last_backward,
                 )
+            elif comp_type == REDUCE_GRAD:
```

Comment:
i'm probably ok with landing as-is for now, but i am wondering if we eventually get away from hardcoding the fsdp-specific logic into the stage.

for autoparallel, we're relying on someone to override stage.perform_reduce_grad, right?